curve25519-dalek/src/montgomery.rs

```diff
@@ -151,6 +151,43 @@ impl MontgomeryPoint {
         Self::mul_base(&s)
     }
 
+    /// Given `self` \\( = u\_0(P) \\), and a big-endian bit representation of an integer
+    /// \\(n\\), return \\( u\_0(\[n\]P) \\). This is constant time in the length of `bits`.
+    ///
+    /// **NOTE:** You probably do not want to use this function. Almost every protocol built on
+    /// Curve25519 uses _clamped multiplication_, explained
+    /// [here](https://neilmadden.blog/2020/05/28/whats-the-curve25519-clamping-all-about/).
+    /// When in doubt, use [`Self::mul_clamped`].
+    pub fn mul_bits_be(self, bits: impl Iterator<Item = bool>) -> MontgomeryPoint {
```

Comment:
Do we care that this doesn't enforce how many bits are passed? ie you can pass an empty iter and it will be equivalent to multiplying by zero, or you can pass a longer iterator and compute a "longer" multiplication

Reply:
Should it use `ExactSizeIterator` and assert len == 256? that would still allow multiplying by something bigger than the curve's order though

Reply:
I do like how it's infallible now.

Perhaps it could have a `debug_assert!`? That wouldn't necessarily require `ExactSizeIterator`, it would just need a sort of postcondition check that it iterated over 256 bits.

Reply:
Why limit the function at all? Multiplication by arbitrary integers is always well-defined. If you don't want to reduce then you're free not to